src/main/java/soot/asm/AsmUtil.java

```diff
@@ -297,7 +286,88 @@ public static String baseTypeName(String s) {
     }
   }
 
-  private AsmUtil() {
+  private AsmUtil() {}
+
+  public static int byteCodeToJavaVersion(int bytecodeVersion) {
+    int javaVersion;
+
+    switch (bytecodeVersion) {
+      case (Opcodes.V1_5):
```

Comment:
wouldn't it be easier to store the java_version directly in the Opcodes and then calling

 `OpCode opcode = OpCodes.valueOf(byteCodeVersion); 
return opcode.get_java_version();`

Reply:
I have to have a look at this. It will only work if the returned java version has the same format as the one defined in the options.  I wouldN#t be suprised if somewhere in soot the version is compared with some constant string which would not change with a format change of the constants in `Options`.

Reply:
@pavanupb Could you please check if the approach proposed by andreas can be used to replace my switch case?

Reply:
> @pavanupb Could you please check if the approach proposed by andreas can be used to replace my switch case?

@mbenz89 Again I took the changes from the develop here. So should be reflecting once my PR is merged.

Reply:
Andreas proposed to change the functionality by using OpCodes class. Could you check if that makes sense, i.e. produces the same String values for the versions.

Reply:
Should I use the statement proposed by andreas in the constructor of AsmUtil class?.

Reply:
No. I think you should replace the switch case here and in `javaToBytecodeVersion` with the code he proposed. Make sure that the result in the same strings first! Maybe write a small test case that  uses the switch cases and the Opcodes approach and compare the results for all the java versions

Reply:
@mbenz89 Which package should I be using for Opcodes. I checked with org.jf.dexlib2.Opcode.valueof() function, but I am getting an exception thrown when I send the bytecodeVersion value as 49.

Reply:
you need to use asm. dexlib is for android.

Reply:
So it should be objectweb.asm correct?.

Reply:
The Opcodes interface does not have a definition for valueOf. Am I using the right package?

Reply:
Yes it shout be object web.sm I think. If there is no value of Andreas maybe mebt a different library. If you cannot get this done under 1h it's not worth the effort.